src/SmartComponents/BaselinesTable/redux/baselinesTableReducer.js

```diff
@@ -61,6 +61,33 @@ export function baselinesTableReducer(state = initialState, action) {
                 }
             });
 
+            return {
+                ...state,
+                baselineTableData: newBaselineTableData,
+                selectedBaselineIds: selectedBaselines
+            };
+        case `${types.SELECT_ONE_BASELINE}`:
+            selectedBaselines = [ ...state.selectedBaselineIds ];
+
+            if (action.payload.id === '') {
+                selectedBaselines = [];
+            } else if (action.payload.isSelected) {
+                selectedBaselines.pop();
+                selectedBaselines.push(action.payload.id);
+            } else if (!action.payload.isSelected) {
+                selectedBaselines.pop();
+            }
+
+            state.baselineTableData.map(row => {
+                if (action.payload.id.includes(row[0])) {
+                    row.selected = action.payload.isSelected;
+                } else {
+                    row.selected = false;
+                }
+
+                newBaselineTableData.push(row);
+            });
```

Comment:
I think if we re-factored the logic in the SELECT_BASELINE reducer to some helpers, we could use one action to handle both cases. Not required for this change and I might take a crack at doing that.

Reply:
Agree, but definitely want to tackle this later. I'll make a card for it so we ensure it gets addressed.